canonical/lxd-ui
# feat: [WD-14512] Proxy device configuration

## Done

- Adding Proxy Configuration...

**Draft PR for Proxy device config**
Current issues / todos:
[] Correct implementation of the tiered Listen / Connect sections given the restrictions of the current components getInheritedDeviceRow and getConfigurationRowBase.
[] Correct / best case implementation of the UI design.

Upcoming:
[] Reviewing create/edit instance/profile pipeline with new api params.
[] Explanation tooltips for disabled buttons
[] NAT boolean logic (If false, connect and listen types must be the same)

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Navigate to the Create Instance page to see Proxy device config upon creating a new instance, or also access from the Create Profile / Edit Instance / Edit Profile pages.

## Screenshots

Pending...
